pkg/sql/codegen/intermediate_representation.go

```diff
@@ -40,17 +40,18 @@ const (
 //
 // IsSparse indicates the type of tensor for a field. True means the tensor is a sparse tensor.
 type FieldMeta struct {
-	Name      string    `json:"name"`      // e.g. "spetal_length"
-	DType     FieldType `json:"dtype"`     // e.g. "float", "int32"
-	Delimiter string    `json:"delimiter"` // e.g. ","
-	Shape     []int     `json:"shape"`     // e.g. [1], [1 2 3]
-	IsSparse  bool      `json:"is_sparse"` // e.g. false
+	Name       string            `json:"name"`       // e.g. "spetal_length"
+	DType      FieldType         `json:"dtype"`      // e.g. "float", "int32"
+	Delimiter  string            `json:"delimiter"`  // e.g. ","
+	Shape      []int             `json:"shape"`      // e.g. [1], [1 2 3]
+	IsSparse   bool              `json:"is_sparse"`  // e.g. false
+	Vocabulary map[string]string `json:"vocabulary"` // use a map to generate a list without duplication
```

Comment:
Maybe add a comment explaining what is a `Vocabulary`?

Reply:
Will do in the next PR.